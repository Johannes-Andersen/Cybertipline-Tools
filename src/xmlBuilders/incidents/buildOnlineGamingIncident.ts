import { XMLBuilder } from 'fast-xml-parser';
import type { OnlineGamingIncident } from '../../types';

const builder = new XMLBuilder({
  attributeNamePrefix: '@_',
  ignoreAttributes: false,
  suppressBooleanAttributes: false,
});

export const buildOnlineGamingIncident = (
  onlineGamingIncident: OnlineGamingIncident,
  keyName = 'onlineGamingIncident',
): string =>
  builder.build({
    [keyName]: {
      gameName: onlineGamingIncident.gameName,
      console: onlineGamingIncident.console,
      content: onlineGamingIncident.content,
      additionalInfo: onlineGamingIncident.additionalInfo,
    },
  });
