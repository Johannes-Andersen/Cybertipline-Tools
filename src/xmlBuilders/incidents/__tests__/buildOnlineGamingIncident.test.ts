import { describe, expect, it } from 'vitest';
import type { OnlineGamingIncident } from '../../../types';
import { buildOnlineGamingIncident } from '../buildOnlineGamingIncident';

describe('buildOnlineGamingIncident', () => {
  it('should build a online gaming incident', () => {
    const onlineGamingIncident: OnlineGamingIncident = {
      console: 'My Console',
      gameName: 'My Game',
      additionalInfo: 'This is a test',
      content: 'This is a test',
    };

    const result = buildOnlineGamingIncident(onlineGamingIncident);

    expect(result).toMatchSnapshot();
  });
});
