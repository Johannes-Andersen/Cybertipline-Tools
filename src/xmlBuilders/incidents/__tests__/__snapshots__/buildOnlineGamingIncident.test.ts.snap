// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildOnlineGamingIncident > should build a online gaming incident 1`] = `"<onlineGamingIncident><gameName>My Game</gameName><console>My Console</console><content>This is a test</content><additionalInfo>This is a test</additionalInfo></onlineGamingIncident>"`;
